### ISSUE TITLE
Support setting the agent log level to Trace in Fleet

### ISSUE THREAD

**Issue by cmacknz** [elastic/elastic-agent#2212]
The Fleet UI does not know about the trace level that was introduced in https://github.com/elastic/elastic-agent/issues/1955

<img width="1392" alt="Screen Shot 2023-01-26 at 11 50 27 AM" src="https://user-images.githubusercontent.com/3466215/214898058-eb3c8ff1-d241-4a1f-be35-82ea57add1bb.png">

We will need to ensure the agent settings action handler understands the new Trace log level:

https://github.com/elastic/elastic-agent/blob/fefe64fff4369bddd806fdccf691ef6213adab42/internal/pkg/agent/application/actions/handlers/handler_action_settings.go#L52-L60

In the Fleet UI the following two definitions need to be changed to allow setting the Trace level.

https://github.com/elastic/kibana/blob/f78236a2e4c1532a9a135444bea95a0f89d1047e/x-pack/plugins/fleet/server/types/models/agent.ts#L32-L42

```typescript
  schema.object({
    type: schema.oneOf([schema.literal('SETTINGS')]),
    data: schema.object({
      log_level: schema.oneOf([
        schema.literal('debug'),
        schema.literal('info'),
        schema.literal('warning'),
        schema.literal('error'),
      ]),
    }),
  })
```

https://github.com/elastic/kibana/blob/f78236a2e4c1532a9a135444bea95a0f89d1047e/x-pack/plugins/fleet/public/applications/fleet/sections/agents/agent_details_page/components/agent_logs/constants.tsx#L48-L53

```typescript
export const AGENT_LOG_LEVELS = {
  ERROR: 'error',
  WARNING: 'warning',
  INFO: 'info',
  DEBUG: 'debug',
};
```

**Comment by sjoukedv** [comment]
In addition to this can we support overriding the default log level per policy? I can only change the log level per agent, but the agents come and go in an autoscaling Kubernetes cluster.

Additionally, we can support setting the log level from the environment in the configuration like the elasticsearch configuration: https://github.com/elastic/elastic-agent/blob/main/elastic-agent.docker.yml#L7-L9. This would allow me to just run the container `docker.elastic.co/beats/elastic-agent` with log level from the environment.

**Comment by defensivedepth** [comment]
Would also like to see the ability to override the default log level per policy or globally.

**Comment by cmacknz** [comment]
Changing the log level per policy needs to be supported in Fleet, it should be possible to do this today for a standalone agent. I've opened https://github.com/elastic/kibana/issues/158861 to track this.

For the trace log level in this issue was used to hide some messages that were generated for every event and flooding the log files, making them useless for debugging. There isn’t any value in turning this on right now.

We can eventually add the ability to change the log level per input or integration, we are unlikely to expose the trace level until we do that as its the only way to control the volume of logs enough to keep it useful.

**Comment by jlind23** [comment]
Depends on https://github.com/elastic/kibana/issues/158861

**Comment by jlind23** [comment]
@cmacknz now that https://github.com/elastic/kibana/issues/158861 landed are we good with closing this one or do we really want to add the trace option in Fleet UI?

**Comment by cmacknz** [comment]
The need for this is going away now that we can write noisy per event logs to a separate file, closing.

**Comment by cmacknz** [comment]
@belimawr where is the event log file for agent documented?

**Comment by belimawr** [comment]
> @belimawr where is the event log file for agent documented?

It's documented alongside the other already existing logs documentation:
- https://www.elastic.co/guide/en/fleet/current/elastic-agent-standalone-logging-config.html (look for `agent.logging.event_data`)
- https://github.com/elastic/elastic-agent/blob/main/docs/elastic-agent-logging.md (look for `agent.logging.event_data`)
- https://github.com/elastic/elastic-agent/blob/a1fd2c9d9a7cc3931f8f1d452a730b6d216f8bd3/elastic-agent.reference.yml#L283-L329 (look for `agent.logging.event_data`)

**Comment by defensivedepth** [comment]
So is `trace` no longer an option? I'm trying to view the raw logs that Elastic Agent (Filebeat) is sending. Normally with Filebeat we would just enable `trace` log level and we can see this kind of logging.

**Comment by belimawr** [comment]
> So is `trace` no longer an option? I'm trying to view the raw logs that Elastic Agent (Filebeat) is sending. In the past with straight Filebeat we would just enable `trace` log level and we can see this kind of logging.

@defensivedepth what logs are you looking for? Is is the raw events Filebeat publishes or something else?

Do you need to be able to see them in Kibana or collecting an Agent diagnostics is an option?

**Comment by defensivedepth** [comment]
@belimawr Yes, raw events Filebeat publishes. Collecting an Agent diagnostics is an option - does that exist already?

**Comment by belimawr** [comment]
Yes, it is!

**Comment by belimawr** [comment]
Yes, it is!

You can run `elastic-agent diagnostics collect`, this will generate a zip file with the diagnostics bundle, there will be a folder in there with all the logs.

Another option is via Fleet UI: Fleet -> Agents -> <your agent name> -> Diagnostics -> Request diagnostics.zip

**Comment by defensivedepth** [comment]
Thanks @belimawr !

**Comment by reyesj2** [comment]
> Yes, it is!
> 
> You can run `elastic-agent diagnostics collect`, this will generate a zip file with the diagnostics bundle, there will be a folder in there with all the logs.
> 
> Another option is via Fleet UI: Fleet -> Agents -> -> Diagnostics -> Request diagnostics.zip

@belimawr I tried this out and it seems I get the same level of logs seen from the fleet UI? I am still unable to view the raw event logs.

**Comment by reyesj2** [comment]
Thanks for the assistance, I am running ```elastic-agent diagnostics collect``` and reviewing log files within the generated zip under ```logs/elastic-agent-*/*.ndjson```.  I do not see the raw events being logged there.

I have the elastic-agent shipping to logstash, so the correct work around I use is to configure an additional logstash pipeline to write events to a file.

**Comment by reyesj2** [comment]
Is there an associated configuration option for logging to events directory?

Here is what is under ./logs in the unzipped diagnostics
```
./logs:
total 4
drwxr-xr-x. 2 root root 4096 Aug 30 17:30 elastic-agent-2df2c1

./logs/elastic-agent-2df2c1:
total 81576
-rw-r--r--. 1 root root 10485851 Aug 30 16:40 elastic-agent-20240830-145.ndjson
-rw-r--r--. 1 root root 10486076 Aug 30 16:49 elastic-agent-20240830-146.ndjson
-rw-r--r--. 1 root root 10486000 Aug 30 16:54 elastic-agent-20240830-147.ndjson
-rw-r--r--. 1 root root 10485614 Aug 30 17:02 elastic-agent-20240830-148.ndjson
-rw-r--r--. 1 root root 10486416 Aug 30 17:08 elastic-agent-20240830-149.ndjson
-rw-r--r--. 1 root root 10486091 Aug 30 17:16 elastic-agent-20240830-150.ndjson
-rw-r--r--. 1 root root 10486297 Aug 30 17:23 elastic-agent-20240830-151.ndjson
-rw-r--r--. 1 root root 10106104 Aug 30 17:30 elastic-agent-20240830-152.ndjson
```

**Comment by belimawr** [comment]
@reyesj2 which version of the Elastic-Agent are you running? The event log file was released in `v8.15.0`

**Comment by reyesj2** [comment]
That's it, I am running version 8.14.3. Thanks for your time on this
